CPAC/pipeline/utils.py

```diff
@@ -17,13 +17,143 @@
 """C-PAC pipeline engine utilities."""
 
 from itertools import chain
+import os
+import subprocess
+
+import nibabel as nib
 
 from CPAC.func_preproc.func_motion import motion_estimate_filter
 from CPAC.utils.bids_utils import insert_entity
+from CPAC.utils.monitoring import IFLOGGER
 
 MOVEMENT_FILTER_KEYS = motion_estimate_filter.outputs
 
 
+def find_pixdim4(file_path):
+    """Find the pixdim4 value of a NIfTI file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the NIfTI file.
+
+    Returns
+    -------
+    float
+        The pixdim4 value of the NIfTI file.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the file does not exist.
+    nibabel.filebasedimages.ImageFileError
+        If there is an error loading the NIfTI file.
+    IndexError
+        If pixdim4 is not found in the header.
+    """
+    if not os.path.isfile(file_path):
+        error_message = f"File not found: {file_path}"
+        raise FileNotFoundError(file_path)
+
+    try:
+        nii = nib.load(file_path)
+        header = nii.header
+        pixdim = header.get_zooms()
+        return pixdim[3]
+    except nib.filebasedimages.ImageFileError as e:
+        error_message = f"Error loading the NIfTI file: {e}"
+        raise nib.filebasedimages.ImageFileError(error_message)
+    except IndexError as e:
+        error_message = f"pixdim4 not found in the header: {e}"
+        raise IndexError(error_message)
+
+
+def update_pixdim4(file_path, new_pixdim4):
+    """Update the pixdim4 value of a NIfTI file using 3drefit.
```

Comment:
Do we have to worry about

> [If the dataset has slice time offsets, these will be scaled by the factor newTR/oldTR. This scaling does not apply if you use '-Tslices' in the same 3drefit run.](https://afni.nimh.nih.gov/pub/dist/doc/program_help/3drefit.html)

?

Reply:
I am not sure about this one. May be @sgiavasis could advise on this.

Reply:
The reason to use AFNI `3drefit` instead of `nibabel` is to change the `Time step` in `3dinfo` as well along with `pixdim4`

Reply:
I looked into this further. It seems that it only refers to, and impacts, an AFNI-specific slice-timing setting in the header, which can be checked using `3dinfo -slice_timing`.

I ran this on several different datasets and found all of them to be unset:
`0.000000|0.000000|0.000000|0.000000|0.000000|0.000000|... etc.`

This was both raw data and final outputs, just to make sure C-PAC wasn't clearing this setting as well.

I am pretty sure this is an esoteric and rarely-used setting. As we know, slice-timing offsets and sequences are now contained in BIDS sidecars. But in the interest in keeping up with the theme of this issue and PR, where we maintain the old header information in case a user wants to run post-processing using some non-BIDS-enabled tools, I would still want to avoid inadvertently changing an AFNI header slice-timing offset to something wrong.

So the question remains, how often is this setting actually populated? Again, I think rarely. But if it is, when a tool invoked by C-PAC clobbers the TR value in the `pixdim4` header field, is it also clobbering this field? I doubt it, and if we run `3drefit -TR` and it scales `newTR/oldTR` like the `3drefit` docs say, it might scale by, let's say, `0.8/1.0` as it would in our recent specific case.

Also curious about `-Tslices` - can this be used blankly with no parameter, to keep it the same, but prevent the scaling? May be worth manually setting this AFNI header field with the offsets from the sidecar, then trying a few of these permutations out.